### PR TITLE
remove bridge API from codegen

### PR DESCRIPTION
**Pull request by bdhirsh**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58065 add a boxed CPU fallback kernel
* #58064 remove xla-specific stuff from codegen (minus CPU fallback)
* #57510 generate inplace/out kernels for xla
* #57361 [codegen] split out backend-specific information from NativeFunction in the model
* **#55796 remove bridge API from codegen**
* #56597 [external codegen] better yaml error messaging, added explicit error message tests
* #55795 add _to_cpu() operator

Differential Revision: [D28474361](https://our.internmc.facebook.com/intern/diff/D28474361)

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -1051,6 +1051,9 @@
 - func: _copy_from(Tensor self, Tensor dst, bool non_blocking=False) -> Tensor
   dispatch: {}
 
+- func: _copy_from_temp(Tensor self, Tensor dst) -> Tensor
```
Comment (ezyang) [inline aten/src/ATen/native/native_functions.yaml:1054]: comment

File: aten/src/ATen/templates/aten_xla_type_default.cpp [hunk 2]
```diff
@@ -14,6 +14,93 @@
 
 namespace torch_xla {
 
+// convenience helpers for extracting out an optional c10::Device
```
Comment (ezyang) [inline aten/src/ATen/templates/aten_xla_type_default.cpp:17]: Don't we have other copies of this functionality floating around other places?

**Review by ezyang — APPROVED** [review]

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (bdhirsh) [reply aten/src/ATen/native/native_functions.yaml:1050]: yep. It needs a better name too 😛 I originally put `_copy_from_out`, but the codegen then flags it as an `out` function.

From what I can tell, the only different between the two is `_copy_from` will fail if you pass in tensors of different sizes, while `_copy_from_temp` will resize correctly and also null out the views on the XLA tensor ([here](https://github.com/pytorch/xla/blob/f41a53458d12b19027f02e5ec34f51d187a63368/torch_xla/csrc/tensor.cpp#L882)). So maybe `_copy_from_and_resize`.

**Comment by bdhirsh** [comment]
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28474361).

**Pull request closed without merging**